opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -186,25 +190,33 @@ namespace Opm
         // Note: currently, the WQTotal definition is still depends on Injector/Producer.
         assert(comp_idx < num_components_);
 
-        if (well_type_ == INJECTOR) { // only single phase injection
-            // TODO: using comp_frac here is dangerous, it should be changed later
-            // Most likely, it should be changed to use distr, or at least, we need to update comp_frac_ based on distr
-            // while solvent might complicate the situation
-            const auto pu = phaseUsage();
-            const int legacyCompIdx = ebosCompIdxToFlowCompIdx(comp_idx);
-            double comp_frac = 0.0;
-            if (has_solvent && comp_idx == contiSolventEqIdx) { // solvent
-                comp_frac = comp_frac_[pu.phase_pos[ Gas ]] * wsolvent();
-            } else if (legacyCompIdx == pu.phase_pos[ Gas ]) {
-                comp_frac = comp_frac_[legacyCompIdx];
-                if (has_solvent) {
-                    comp_frac *= (1.0 - wsolvent());
+        if (this->isInjector()) { // only single phase injection
+            double inj_frac = 0.0;
+            switch (this->wellEcl().injectorType()) {
+            case Well2::InjectorType::WATER:
+                if (comp_idx == FluidSystem::waterCompIdx) {
```

Comment:
Maybe move these lines to a method in the interface that returns inj_frac based on comp_idx